File: July_2023/4_July_23/137.cpp
class Solution {
public:
    int singleNumber(vector<int>& nums) {
        int size = nums.size();
        map<int,int>mp;
        for(int i=0;i<size;i++)
        {
            mp[nums[i]]++;
        }
        int ans;
        for(auto element:mp)
        {
            if(element.second==1)
            {
                ans=element.first;
            }
        }
        return ans;

    }
};

File: July_2023/4_July_23/137.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public int singleNumber(int[] nums) {
        int size = nums.length;
        Map<Integer, Integer> mp = new HashMap<>();
        for (int i = 0; i < size; i++) {
            int num = nums[i];
            mp.put(num, mp.getOrDefault(num, 0) + 1);
        }
        int ans = 0;
        for (Map.Entry<Integer, Integer> entry : mp.entrySet()) {
            if (entry.getValue() == 1) {
                ans = entry.getKey();
                break;
            }
        }
        return ans;
    }
}
